app/controllers/all-events.js

```diff
@@ -0,0 +1,320 @@
+                for (let i = 0; i < events.length; i++) {
+                    sorted.push(events[i]);
+                }
+                $(item.target).addClass("toggleclick");
+            }
+            this.set("model.eventShown", sorted);
+        },
+
+        sortByLocation(item) {
+            let events = this.get("model.eventShown");
+            let sorted = [];
+            if ($(item.target).hasClass("toggleclick")) {
+                for (let i = 0; i < events.length; i++) {
+                    console.log("hi");
+                    sorted.push(events[events.length - 1 - i]);
+                    $(item.target).removeClass("toggleclick");
+                }
+            } else {
+                events.sort(function(a, b) {
+                    if (a.Location < b.Location) {
+                        return -1;
+                    }
+                    if (a.Location > b.Location) {
+                        return 1;
+                    }
+                    return 0;
+                });
+                for (let i = 0; i < events.length; i++) {
+                    sorted.push(events[i]);
+                }
+                $(item.target).addClass("toggleclick");
+            }
+            this.set("model.eventShown", sorted);
+        },
+
+        SortByDate(item) {
+            let events = this.get("model.eventShown");
+            let sorted = [];
+            if ($(item.target).hasClass("toggleclick")) {
+                for (let i = 0; i < events.length; i++) {
+                    console.log("hi");
+                    sorted.push(events[events.length - 1 - i]);
+                    $(item.target).removeClass("toggleclick");
+                }
+            } else {
+                events.sort(function(a, b) {
+                    if (a.timeObject < b.timeObject) {
+                        return -1;
+                    }
+                    if (a.timeObject > b.timeObject) {
+                        return 1;
+                    }
+                    return 0;
+                });
+                for (let i = 0; i < events.length; i++) {
+                    sorted.push(events[i]);
+                }
+                $(item.target).addClass("toggleclick");
+            }
+            this.set("model.eventShown", sorted);
+        },
+
+        toggleRequestSent(item) {
+
+
+
+            let lastForm = this.get("lastForm");
+
+            if (lastForm && $(item.target).hasClass("onclickOrange")) {
+                let elements = $(lastForm).nextAll();
+                for (var i = 0; i < 3; i++) {
+                    if ($(window).innerWidth() < 768) {
+                        $(elements[i]).slideUp();
+                        console.log("twice");
+                    }
+                }
+                let form = $(item.target);
+                this.set("lastForm", form);
+            } else {
+                if (lastForm) {
+                    $(lastForm).removeClass("onclickOrange");
```

Comment:
What is this class you are using.  its not on any elements in your handlebars file.  If its just for the css style there might be a better way to do that. (not 100% sure)